### PR TITLE
Pass service port to SPAAS service

### PR DESCRIPTION
**Pull request by rakanalh**
Introduce a new CLI param to be able to specify the port
for the SPAAS service

### CODE REVIEW

**Comment by rakanalh** [comment]
The reason we're doing this is that a `Address In Use` exception is raised when running multiple SPs on the same machine. Regardless of the exception, the SPs continue to run the scenarios as expected.

The PR is a PoC on trying to prevent raising an exception. However, as you're suggesting a different approach, i wouldn't mind closing this PR and implementing this at some point.

File: scenario_player/main.py [hunk 1]
```diff
@@ -228,18 +236,27 @@ def run(
...
+        }
+    }
+    scenario_definition = ScenarioDefinition(scenario_file, data_path, overrides)
+
     # Dynamically import valid Task classes from sceanrio_player.tasks package.
     collect_tasks(tasks)
 
     # Start our Services
-    service_process = ServiceProcess()
+    service_process = ServiceProcess(port=service_port)
```
Comment (rakanalh) [inline scenario_player/main.py]: @nlsdfnbch Not sure i get your comment. You mean i should be using the `scenario_definition.spaas.rpc.port` here?

Comment (nlsdfnbch) [reply scenario_player/main.py]: Yupp, that's what I'm saying.

**Comment by nlsdfnbch** [comment]
> The reason we're doing this is that a `Address In Use` exception is raised when running multiple SPs on the same machine. Regardless of the exception, the SPs continue to run the scenarios as expected.
> 
> The PR is a PoC on trying to prevent raising an exception. However, as you're suggesting a different approach, i wouldn't mind closing this PR and implementing this at some point.

Unless you have dedicated accounts for each run of the sp, running them concurrently wouldn't work anyway. 

If you do have dedicated accounts, you could alternatively run them in a docker container, which should work.

File: scenario_player/main.py (hunk 1 above)
Comment (rakanalh) [reply scenario_player/main.py]: Done.

**Comment by rakanalh** [comment]
btw, there's a false positive on CI because a lot of unit tests are failing for me locally.

**Pull request closed without merging**